src/components/app/SourceFetcher.js

```diff
@@ -59,52 +60,50 @@ class SourceFetcherImpl extends React.PureComponent<Props> {
       beginLoadingSourceFromUrl,
       finishLoadingSource,
       failLoadingSource,
+      symbolServerUrl,
+      profile,
     } = this.props;
 
-    const errors: SourceLoadingError[] = [];
-    const parsedName = parseFileNameFromSymbolication(file);
-    const downloadRecipe = getDownloadRecipeForSourceFile(parsedName);
-
-    // First, try to fetch just the single file from the web.
-    switch (downloadRecipe.type) {
-      case 'CORS_ENABLED_SINGLE_FILE': {
-        const { url } = downloadRecipe;
-        beginLoadingSourceFromUrl(file, url);
-
-        try {
-          const response = await fetch(url, { credentials: 'omit' });
-
+    const addressProof =
+      profile !== null ? findAddressProofForFile(profile, file) : null;
+
+    const fetchSourceResult = await fetchSource(
+      file,
+      symbolServerUrl,
+      addressProof,
+      {
+        fetchUrlResponse: async (url: string, postData?: MixedObject) => {
+          beginLoadingSourceFromUrl(file, url);
+
+          const requestInit =
+            postData !== undefined
+              ? {
+                  body: postData,
+                  method: 'POST',
+                  mode: 'cors',
```

Comment:
Shouldn't we also put `credentials: 'omit'` in this case?

Reply:
I'm not fully familiar with the meaning of these parameters, to be honest. I cargo-culted this code from the symbolication POST request which also only has `mode: 'cors'` and no `credentials: 'omit'`. I'll do some reading, to see if the combination of those two things is even meaningful, or if one subsumes the other.

Reply:
It sounds like the two are indeed independent. I'll add `credentials: 'omit'` because we never want to send cookies.

Reply:
Actually I'm reading in the Fetch spec that `mode: 'cors'` is the default when the first argument is a string, so it could be omitted.
There's no problem in keeping it explicitely, but then it looks like from the source code that the GET case doesn't use CORS where actually it does as soon as we request cross-origin (even for GET requests).